### PR TITLE
chore: wire up EKS bootstrapper and AWSMachine controller for AWSManagedControlPlane

### LINKED ISSUE #1897: AWSMachine controller should support AWSManagedControlPlanes
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
The EKS controlplane and bootstrap providers are both merged. Next, we need the AWSMachine controller to create ManagedControlPlaneScopes instead of ClusterScopes when appropriate. 

This will complete the circle for creating and joining EKS nodes.

### PR DESCRIPTION
**Pull request by rudoi**
**What this PR does / why we need it**: Wires up AWSMachine controller to join AWSManagedControlPlanes. Also checks for EKSClusterName in the EKS bootstrap provider.

With this PR, I was able to fully join a MachineDeployment to the cluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1897

### CODE REVIEW

**Comment by rudoi** [comment]
Doing a small refactor to get rid of some ugly nesting, then the linters should pass.

File: bootstrap/eks/main.go [hunk 1]
```diff
@@ -34,8 +34,9 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 
-	eksbootstrapv1alpha3 "sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/api/v1alpha3"
-	eksbootstrapcontrollers "sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/controllers"
+	bootstrapv1 "sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/api/v1alpha3"
+	bootstrapv1controllers "sigs.k8s.io/cluster-api-provider-aws/bootstrap/eks/controllers"
+	expinfrav1 "sigs.k8s.io/cluster-api-provider-aws/exp/api/v1alpha3"
```
Comment (richardcase) [inline bootstrap/eks/main.go:39]: We've used `infrav1exp` in the control plane side but thats fine.

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 2]
```diff
@@ -156,6 +159,10 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, log logr.Logger, c
 		}
 	}
 
+	if cluster.Spec.ControlPlaneRef == nil || cluster.Spec.ControlPlaneRef.Kind != "AWSManagedControlPlane" {
+		return ctrl.Result{}, errors.New("Cluster's controlPlaneRef needs to be an AWSManagedControlPlane in order to use the EKS bootstrap provider")
```
Comment (richardcase) [inline bootstrap/eks/controllers/eksconfig_controller.go:163]: This error could be a static package level error to help with `goerr113`. But thats not essential for this.

**Review by richardcase — APPROVED** [review]

**Comment by richardcase** [comment]
For final approval:

/assign @randomvariable

File: bootstrap/eks/main.go (hunk 1 above)
Comment (rudoi) [reply bootstrap/eks/main.go:39]: I'm interested in doing a clean up PR that removes a handful of redundancies / cleans up some standardizations. For example I think we're parsing version strings with different helpers in the control plane and bootstrap stuff. That sound okay?

**Pull request merged**